src/Model/InputParser.js

```diff
@@ -0,0 +1,60 @@
+import { DEFAULT_DIVIDERS, ERROR_MESSAGES } from "../constants.js";
+
+class InputParser {
+  parse(input) {
+    if (input === "") {
+      return [0];
+    }
+
+    const { divider, numberString } = this.parseInput(input);
+    const numberArray = this.createNumberArray(numberString, divider);
+    return this.validateNumber(numberArray);
+  }
+
+  parseInput(input) {
+    if (input.startsWith("//")) {
+      return this.parseCustomDivider(input);
+    }
+
+    const defaultDividers = DEFAULT_DIVIDERS;
+    if (defaultDividers.some((div) => input.includes(div))) {
+      return { divider: defaultDividers, numberString: input };
+    }
+
+    throw new Error(ERROR_MESSAGES.NO_VALID_DIVIDER);
+  }
+
+  parseCustomDivider(input) {
+    const endIndex = input.indexOf("\\n");
+    const customDivider = input.slice(2, endIndex);
+    const numberString = input.slice(endIndex + 2);
+
+    if (numberString.length === 0) {
+      throw new Error(ERROR_MESSAGES.NO_NUMBERS_INPUT);
+    }
+
+    return { divider: [customDivider], numberString };
+  }
+
+  createNumberArray(numberString, divider) {
+    divider.forEach((divider) => {
+      numberString = numberString.split(divider).join(",");
```

Comment:
```suggestion
      let splitString = numberString; // createNumberArray 메서드 위쪽에서 선언
      splitString = numberString.split(divider).join(",");
```
인자로 들어온 `numberString`의 값을 바꾸고 있는데 변수를 하나 만들어서 기존 인자에 대한 불변성을 지키는 것은 어떻게 생각하시나요??

++ 추가로 지금 방식은 구분자들을 순회하면서 문자열을 나누고 다시 문자열로 변환하고 있는데 정규 표현식으로 구분자들을 합하면 구분자들을 순회하지 않고도 문자열을 나눌 수 있을 것 같아요!
```js
const dividerRegExp = new RegExp(dividers.join("|"), "g"); // 구분자들을 합하고 정규표현식으로 변환
const numbers = numberString.split(dividerRegExp).map(Number); // 변환한 구분자를 기준으로 문자열 나누기
```